rasa/core/actions/forms.py

```diff
@@ -259,28 +259,22 @@ def extract_other_slots(
         """
         slot_to_fill = tracker.get_slot(REQUESTED_SLOT)
 
-        entity_type_of_slot_to_fill = self._get_entity_type_of_slot_to_fill(
```

Comment:
entity type of slot to fill matters. It checks for that when filling other slots with different roles and groups inside `self.entity_is_desired`

Reply:
So in `extract_other_slots` you only want to fill other slots if their entity type is the same as the requested slot?
If so, why bother calling it when `REQUESTED_SLOT` is none?
In addition, it looks a little bug prone in case the two entity types are different, though I am probably overlooking something.

Reply:
> fill other slots if their entity type is the same as the requested slot

or when their `from_entity` mapping is unique

> If so, why bother calling it when REQUESTED_SLOT is none?

because even if entity name is the same as the slot, we still need to extract the value for validation

Reply:
> it looks a little bug prone in case the two entity types are different

what do you mean?

Reply:
WellI I can see the case is caught in `entity_is_desired`, so it's not a problem, just a little late.
But in case of filling from the triggering intent, wouldn't you want to fill other slots if a unique mapping exists? At the moment `entity_type_of_slot_to_fill` would be `None` and thus `entity_is_desired` return `False`.

Reply:
> wouldn't you want to fill other slots if a unique mapping exists?
that's we do in the master

Reply:
> > wouldn't you want to fill other slots if a unique mapping exists?
> 
> that's we do in the master

Cool, I have to try that next week, it's basically midnight here.